analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/pom/DataExtractor.java

```diff
@@ -0,0 +1,395 @@
+
+import eu.fasten.analyzer.pomanalyzer.pom.data.Dependency;
+import eu.fasten.analyzer.pomanalyzer.pom.data.DependencyData;
+import eu.fasten.analyzer.pomanalyzer.pom.data.DependencyManagement;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.dom4j.DocumentException;
+import org.dom4j.Node;
+import org.dom4j.io.SAXReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataExtractor {
+
+    private List<String> mavenRepos;
+    private static final Logger logger = LoggerFactory.getLogger(DataExtractor.class);
+
+    private String mavenCoordinate = null;
+    private String pomContents = null;
+
+    public DataExtractor() {
+        this.mavenRepos = Collections.singletonList("https://repo.maven.apache.org/maven2/");
+    }
+
+    /**
+     * Extracts repository URL from POM of certain Maven coordinate.
+     *
+     * @param groupId    groupId of the coordinate
+     * @param artifactId artifactId of the coordinate
+     * @param version    version of the coordinate
+     * @return Extracted repository URL as String
+     */
+    public String extractRepoUrl(String groupId, String artifactId, String version) {
+        String repoUrl = null;
+        try {
+            var scm = extractScm(groupId, artifactId, version);
+            if (scm != null) {
+                var url = scm.selectSingleNode("./*[local-name()='url']");
+                repoUrl = url.getText();
+            }
+        } catch (FileNotFoundException | DocumentException e) {
+            logger.error("Error parsing POM file for: "
+                    + groupId + ":" + artifactId + ":" + version);
+        }
+        return repoUrl;
+    }
+
+    /**
+     * Extracts commit tag from POM of certain Maven coordinate.
+     *
+     * @param groupId    groupId of the coordinate
+     * @param artifactId artifactId of the coordinate
+     * @param version    version of the coordinate
+     * @return Extracted commit tag representing certain version in repository
+     */
+    public String extractCommitTag(String groupId, String artifactId, String version) {
+        String commitTag = null;
+        try {
+            var scm = extractScm(groupId, artifactId, version);
+            if (scm != null) {
+                var tag = scm.selectSingleNode("./*[local-name()='tag']");
+                commitTag = tag.getText();
+            }
+        } catch (FileNotFoundException | DocumentException e) {
```

Comment:
Those exceptions signify different classes of errors, right? Why lump them together?

Reply:
Good point. I will separate them.